commit/types.go

```diff
@@ -159,8 +160,12 @@ type ConsensusObservation struct {
 
 // GasPricesArray returns a list of gas prices
 func (co ConsensusObservation) GasPricesArray() []cciptypes.GasPriceChain {
+	chains := maps.Keys(co.GasPrices)
+	sort.Slice(chains, func(i, j int) bool { return chains[i] < chains[j] })
+
 	gasPrices := make([]cciptypes.GasPriceChain, 0, len(co.GasPrices))
-	for chain, gasPrice := range co.GasPrices {
+	for _, chain := range chains {
+		gasPrice := co.GasPrices[chain]
 		gasPrices = append(gasPrices, cciptypes.NewGasPriceChain(gasPrice.Int, chain))
 	}
```

Comment:
small nit: I think you'd save one line of code and be consistent with the other sorts if you sort after the original loop?

```golang
for chain, gasPrice := range co.GasPrices {
  gasPrices = append(gasPrices, cciptypes.NewGasPriceChain(gasPrice.Int, chain))
}
sort.Slice(gasPrices, func() { ... })
```